### DISCUSSION TITLE
McScan tracks

### DISCUSSION THREAD

**Discussion by srobb1** [GMOD/jbrowse-components#3634] [Q&A] [answered]
Hi!
How do I add mcscan tracks in the config? When i copy the json formated block from the documentation and add to the config I  get a message about the MCScanAnchorsAdapter plugin is not installed. I have the two genome aligned with minimap and have the paf file in the config, and that works. I can make everything work if i add a "new file" in the new session form. But I want users to be able to just click and view.

Thanks,
Sofia

**Comment by cmdcolin** [comment]
(transferred to the 'jbrowse-components' discussion forum)

Hello :)

>When i copy the json formated block from the documentation and add to the config I get a message about the MCScanAnchorsAdapter plugin is not installed

hmm, do you know what version of jbrowse 2 you have, or can you paste what snippet you copied? that sounds odd

In recent versions, we added ability to load more of our synteny types via the CLI

Here is an example of adding a MCScan type file. make sure to update the jbrowse CLI (re-run `npm install -g @jbrowse/cli` for example) and then use

```
jbrowse add-track grape_vs_peach.anchors --bed1 grape.bed --bed2 peach.bed --load copy -a grape,peach

```

also I just would add it was a very good question that you asked during our PAG workshop about whether minimap2 is the best method or just the easiest! On some level, minimap2 was just the easiest for us to demo, and it likely is very good but can require tuning that is outside the scope of our docs...I think we just try to support multiple options so that however people generate their alignments, we can load them and the MCScan style 'protein alignment'-based methods are likely very good for many purposes

**Reply by cmdcolin** [reply]
just as another note, I have an interest in creating more 'tutorial' style docs that load different synteny types (e.g. here is a screenshot from our gallery, here is how you get there from scratch)  so hopefully that will help too

**Reply by jhumann** [reply]
I used the CLI command to load my MCScan anchors and it loaded, but I had to go add the path to the files in the uri line.  Even though I provided the file path (e.g. data/synteny<filename>) it just kept the file name when creating the uri line.  

Do the MCScan tracks have the connecting lines between the genomes like the minimap analysis?  I don't see those nor anything in the anchors tracks.  Having screenshots of what it's supposed to look like in JB2 would be very helpful.

**Reply by cmdcolin** [reply]
@jhumann the mcscan data will show connecting lines, the only thing it won't show is a "base level alignment" since it just shows the connection between genes at the "protein level" so it just connects the gene regions.

example screenshot from grape vs peach https://jbrowse.org/code/jb2/v3.0.3/?config=test_data%2Fconfig_dotplot.json&session=share-9lXm-DpeFs&password=2pZzI

<img width="1205" alt="Screenshot 2025-02-18 at 3 02 29 PM" src="https://github.com/user-attachments/assets/099d94bf-f2da-4377-9610-9efab930cd88" />


the most common reason for "nothing to be shown" is when the assembly names are reversed

also feel free to make a new issue for the URI line thing

**Comment by srobb1** [comment, marked as answer]
Hi Colin,

I have 2.4.2. I updated last week I think by running the `jbrowse upgrade` command. Do I still need to run `npm install -g @jbrowse/cli` ? If so, I need to get IT to do it cause I don't have permissions to run that on my machine. :( 

Here is the snippet that i used as a model and modified with my file and assembly names:
```
{
  "type": "MCScanAnchorsAdapter",
  "mcscanAnchorsLocation": {
    "uri": "https://s3.amazonaws.com/jbrowse.org/genomes/synteny/grape.peach.anchors.gz"
  },
  "bed1Location": {
    "uri": "https://s3.amazonaws.com/jbrowse.org/genomes/synteny/grape_vs_peach/grape.bed.gz"
  },
  "bed2Location": {
    "uri": "https://s3.amazonaws.com/jbrowse.org/genomes/synteny/grape_vs_peach/peach.bed.gz"
  },
  "assemblyNames": ["grape", "peach"]
}
```

**Reply by cmdcolin** [reply]
that looks about right, and then what exactly is the error you get when you use that?

**Comment by srobb1** [comment]
I am guessing that I need to rerun the npm install because I get this when trying to load my anchors file. I will send a request to IT.

 ›   Error: Unexpected arguments: --bed1, /usr/local/data/cca1/cca1.bed, --bed2, /usr/local/data/cca1/chicken.bed
 ›   See more help with --help

**Reply by cmdcolin** [reply]
you may be able to use jbrowse CLI without admin access with `npx @jbrowse/cli add-track ...`  (you may also use `npx @jbrowse/cli@latest add-track ...` to fully ensure that npx downloads the latest every time, sometimes it uses a cached version). the npx stuff is briefly mentioned here https://jbrowse.org/jb2/docs/cli/#installation
